src/components/bars/SSIHeaderBar/index.tsx

```diff
@@ -26,22 +26,24 @@ import {ButtonIconsEnum, HeaderMenuIconsEnum, IHeaderMenuButton, IUser, MainRout
 import SSIProfileIcon from '../../assets/icons/SSIProfileIcon';
 import SSIDropDownList from '../../dropDownLists/SSIDropDownList';
 
-interface Props extends NativeStackHeaderProps {
+export interface HeaderBarProps extends NativeStackHeaderProps {
   headerSubTitle?: string;
   showBorder?: boolean;
   showBackButton?: boolean;
   moreActions?: Array<IHeaderMenuButton>;
   showProfileIcon?: boolean;
+  onBack?: () => void | Promise<void>;
 }
 
 // TODO fix that there is a slight flash of elements moving when navigating
-const SSIHeaderBar: FC<Props> = (props: Props): JSX.Element => {
+// NK: Probably has todo with us passing in a new headerbar element via props on every screen, causing a rerender of the entire dom
+const SSIHeaderBar: FC<HeaderBarProps> = (props: HeaderBarProps): JSX.Element => {
   const {showBorder = false, showBackButton = true, showProfileIcon = true, moreActions = []} = props;
   const dispatch = useDispatch();
   const {showProfileMenu, setShowProfileMenu, showMoreMenu, setShowMoreMenu} = useContext(OnTouchContext);
 
   const onBack = async (): Promise<void> => {
-    props.navigation.goBack();
+    typeof props.onBack === 'function' ? await props.onBack() : props.navigation.goBack();
```

Comment:
Yeah, but in the latter it is not immediately clear that the callback can both come from props and props.navigation which can cause confusion

The main reason is that I do not want to think about what onBack might return when using nullish coalescence. If the call would return null it would also call the 2nd argument. In this case It is a promise so it won't happen, but indeed it is just a defensive approach, which I will always favor in situations like these